examples/next-openai-pages/pages/chat/stream-chat-files/index.tsx

```diff
@@ -0,0 +1,112 @@
+    });
+
+  const [attachments, setAttachments] = useState<FileList | undefined>(
+    undefined,
+  );
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="flex flex-col p-2 gap-2">
+        {messages.map(message => (
+          <div key={message.id} className="flex flex-row gap-2">
+            <div className="w-24 text-zinc-500 flex-shrink-0">{`${message.role}: `}</div>
+
+            <div className="flex flex-col gap-2">
+              {message.content}
+
+              <div className="flex flex-row gap-2">
+                {message.experimental_attachments?.map((attachment, index) =>
+                  attachment.mimeType?.includes('image/') ? (
+                    <img
+                      key={`${message.id}-${index}`}
+                      className="w-24 rounded-md"
+                      src={attachment.url}
+                      alt={attachment.name}
+                    />
+                  ) : attachment.mimeType?.includes('text/') ? (
+                    <div className="w-32 h-24 rounded-md text-xs ellipsis overflow-hidden p-2 text-zinc-500 border">
+                      {atob(attachment.url.split(',')[1])}
+                    </div>
+                  ) : null,
+                )}
+              </div>
+            </div>
+          </div>
+        ))}
+      </div>
+
+      <form
+        onSubmit={event => {
+          handleSubmit(event, {
+            experimental_attachments: attachments,
+          });
+          setAttachments(undefined);
+
+          if (fileInputRef.current) {
+            fileInputRef.current.value = '';
+          }
+        }}
+        className="flex flex-col gap-2 fixed bottom-0 p-2 w-full"
+      >
+        <div className="flex flex-row gap-2 fixed right-2 bottom-14 items-end">
+          {attachments
+            ? Array.from(attachments).map(attachment => {
+                const { type } = attachment;
+
+                if (type.startsWith('image/')) {
+                  return (
+                    <div key={attachment.name}>
+                      <img
+                        className="w-24 rounded-md"
+                        src={URL.createObjectURL(attachment)}
+                        alt={attachment.name}
+                      />
+                      <span className="text-sm text-zinc-500">
+                        {attachment.name}
+                      </span>
+                    </div>
+                  );
+                } else if (type.startsWith('text/')) {
+                  return (
+                    <div
+                      key={attachment.name}
+                      className="w-24 text-zinc-500 flex-shrink-0 text-sm flex flex-col gap-1"
+                    >
+                      <div className="w-16 h-20 bg-zinc-100 rounded-md" />
+                      {attachment.name}
+                    </div>
+                  );
+                }
```

Comment:
what is behavior if the condition is not met?

Reply:
it wouldn't render any previews for file uploads if the condition is not met, figured this would be dependent on how developers want to preview different files